app/controllers/financial_transactions_controller.rb

```diff
@@ -1,63 +1,56 @@
 # frozen_string_literal: true
 class FinancialTransactionsController < ApplicationController
-  before_action :set_financial_transaction, only: [:show, :edit]
-
   after_action :set_return_url, only: [:index]
 
-  # GET /financial_transactions
   def index
-    @q = FinancialTransaction.search(params[:q])
+    # workaround for unknown Ransack bug
```

Comment:
The regular Ransack syntax `FinancialTransaction.search(params[:q])` was not working, which is why filter on the index page wasn't working.  For some reason, Ransack was not actually searching for anything as if `params[:q]` was `nil`.

Reply:
Ransack usually doesn't return anything when params[:q] is nil, that's correct behavior as far as I can tell. Some people choose to apply their own filter when params[:q] is nil like below:
```
def index
  @search = Job.search(params[:q])
  @results = @search.result
  @results = @results.where(:your_date => nil) unless params[:q]
  @job = @results.paginate(:per_page => 10, :page => params[:page])
end
``` 
but I found a better example which actually changes the default return of the search object which may be a better approach:
```
def index
  @search = Job.search(params[:q])
  @search.status_cont = 'Open' unless params[:q]
  @results = @search.result
  @job = @results.paginate(:per_page => 10, :page => params[:page])
end
```
Hope this helps. Both of these can be seen [here](https://stackoverflow.com/questions/13284172/setting-default-search-parameter-on-ransack-for-rails)

Reply:
The issue here was not that `params[:q]` was nil.  It was behaving as though `params[:q]` was nil.
If `params[:q]` is nil, Ransack doesn't do any filtering so it returns everything, which is why the filter function on the index page wasn't working.

Reply:
Also, 
```
if params[:q]
  FinancialTransaction.search(transactable_type_eq: params[:q][:transactable_type])
...
```
looks to be logically equivalent to what this snippet is accomplishing
```
@results = @search.result
@results = @results.where(:your_date => nil) unless params[:q]
```
So there shouldn't be a problem.

Reply:
It largely is, but I suggested it an an improvement because it doesn't require a weird else statement where you pass `nil` to FinancialTransaction.search

Reply:
What do you think about the below implementation? It's a bit more complicated but is a lot cleaner

Reply:
Nice code man, I like that a lot. Thanks for changing it :+1: As long as it works then lgtm